### DISCUSSION TITLE
Anyway to tell what a crash is about?

### DISCUSSION THREAD

**Discussion by Hipshot-zz** [Garux/netradiant-custom#187] [General]
Is there a crash dump or log file somewhere? I think I have two crashes here (need to confirm) in two completely different situations. One is when I model browse my Q3 folder (it's a shit load of models in there since 2005 kinda) and one is when I try to texture browse a folder with pngs and some other custom files that is probably not readable by this editor, or it's the size and/or dimensions of the pngs (doubt that though).

**Comment by xanhast** [comment]
depends on OS, but i think you can try running from a terminal / cmd to get some crash reports

**Comment by Garux** [comment]
Normal log may be found in `radiant\settings\1.6.0\radiant.log`.
I have not found *smooth* way to have stacktrace or memory dump in windows with mingw build system yet.
I'll build debug version with address sanitizer for you later to try to catch these errors.
Wild assumption is VRAM usage overflow, if your journey through folders was too long.

**Comment by Hipshot-zz** [comment]
VRAM seems like such a strange error, or well, sure maybe the model folder then, but it's not 60000 lose files, there's plenty of folders in there. I was first thinking that maybe some model format or some error in some model may cause this.

(Small note: I tried to load an older level that contains a lot of models, that level crashes too, without me even touching the browser, this level was build in 1.4 and that has no issue loading this level and all assets.)


The texture folder issue must be something else, TB has no issues loading this folder nor GTK1.6, would be nice to know what's the difference here (most other textrue folders work, but not all, they all contain the same kind of data).

I will be glad to test any debug version or whatever I can test. I work in W7 and W11 and also have a W10 comp, if this matters at all.

**Comment by Hipshot-zz** [comment]
This is an error from the log when it comes to the browser.

Loaded Model: "models/crack.ase"
Line 0: Unable to parse long: unexpected EOL [#1]
Line 6: Out of range: submaterial index is too large

BUT, this model loads perfectly fine in radiant 1.4, I assume NRC has a different way of loading ase models perhaps? The model itself isn't very advanced.

If I remove this model I get the same error on some other model and I will just assume that if I get rid of that it will find another one deeper down in the structure.

This is the model, if you want to look at it: https://zfight.com/misc/files/crack.ase
This is the same one open in 1.4: https://i.ibb.co/GcTk8PC/image.png

Looks to be the submaterial name, but why would NRC care so much?


---------


The texture issue seems a bit more advanced, I'm really not sure why. It's cant be that the memory runs out, because it's only like 20 small pngs and there's even another folder with a huge png that is like 5000x10000 that works. I have tried to delete pngs, eventually the folder works, but it's not because one specific png and it's not because some of them are not power of two either.

The log just gives me this: 

Using OpenGL font "Myriad Pro,8,-1,5,50,0,0,0,0,0"
OpenGL window configuration: version: 4.6 RGBA: 8888 depth: 24 swapInterval: 1 samples: 0
OpenGL window configuration: version: 4.6 RGBA: 8888 depth: 24 swapInterval: 1 samples: 0
Showing 0 shaders.
Loaded Texture: "textures/stone/mosaic_symbol_1"
Loaded Texture: "textures/stone/stone_10"
Loaded Texture: "textures/stone/stone_10_small"

That's not saying much, so there's something else here, some other issue.

**Reply by Garux** [reply]
NRC is using assimp lib for model loading, ase loader there is more conform to format specs than in picomodel, so it's not chewing any screwed custom export result. Crash should not happen though.

If the texture issue is about concrete folder, it's easier to share it, so i could try on my side and debug.

**Reply by Hipshot-zz** [reply]
Anyway I can PM you the texture file link or mail the link, I don't want to post it publicly, not this one since it's not really a mod.

**Reply by Garux** [reply]
I'd prefer PM in discord, user id 246624639655215114

**Reply by Garux** [reply]
So ase crash is because `NUMSUBMTLS` is not specified in the model
this leads to access of index -1 of empty array
https://github.com/Garux/netradiant-custom/blob/d4e1be39ad14f95ef7cca6860a79b424fc3f7016/libs/assimp/code/AssetLib/ASE/ASEParser.cpp#L629
same would happen with missing `MATERIAL_COUNT` https://github.com/Garux/netradiant-custom/blob/d4e1be39ad14f95ef7cca6860a79b424fc3f7016/libs/assimp/code/AssetLib/ASE/ASEParser.cpp#L477

I believe intended design is to specify sizes beforehand. Btw if you search in picomodel, none of these keywords are found.

**Reply by Garux** [reply]
Thank you for report, fixed ASE crash https://github.com/Garux/netradiant-custom/commit/648a836c5ca79d769442fd9fb9664ed77ff3c1d1
And created PR in assimp repo https://github.com/assimp/assimp/pull/5559

**Comment by Hipshot-zz** [comment]
I see, well, these are old models exported from sketchup, I can't really change them and I can't remove them either. Actually, I probably can't open my latest "big" level in NRC because of it, but that's released and done with so.
This is the source for that map: https://www.quake3world.com/forum/viewtopic.php?f=10&t=50631 

Anyway, I haven't really a need for the browser, I know what I need when I need it and that old model is not needed in the future =)

I will see about getting you the textures, I don't use discord, but I will figure something out.

**Reply by Hipshot-zz** [reply]
I just blured the files a bit, that should do it. The sky folder works well, the generic folder crashes:

https://zfight.com/misc/files/NRCtest.zip

**Reply by Garux** [reply]
This is crash on 16bit roof_1.png, which was fixed recently https://github.com/Garux/netradiant-custom/commit/f3ffdf9f287f40dc115071bc0e84576617923497 (no public build yet).

**Comment by Hipshot-zz** [comment]
Hey, that solved a lot. Thanks for letting me know. I'm kinda glad to find out about it, because now I can get rid of those 16bit images anyway which should really be 8bits so. Thanks again.
